install/helm/agones/templates/extensions.yaml

```diff
@@ -38,7 +38,7 @@ spec:
         {{- if .Values.agones.controller.generateTLS }}
   caBundle: {{ b64enc $ca.Cert }}
         {{- else }}
-  caBundle: {{ .Files.Get "certs/server.crt" | b64enc }}
+  caBundle: {{ default (.Files.Get "certs/server.crt") .Values.agones.controller.tlsCert | b64enc }}

```

Comment:
Would it work to put "certs/server.crt" into values.yaml and just use the new variable here?

From https://helm.sh/docs/chart_template_guide/functions_and_pipelines/:

> In an actual chart, all static default values should live in the values.yaml, and should not be repeated using the default command (otherwise they would be redundant).

I'm not particularly familiar with the `--set-file` option, but it looks like if the new parameter is a file name and we keep the `.Files.Get` here referencing the variable, then it might work (but using the `--set` flag instead of `--set-file` because you wouldn't pass the file contents into the variable).

Reply:
The new parameter expects the file's contents.  The `--set-file` option will set a value to the contents of the file at the provided path - it's equivalent to `--set value=$(cat value.txt)`.

The idea behind the new parameter is to allow the user to pass the certificate directly as a value.  When the value isn't provided, `.Files.Get` will be used as a fallback to read the file from the `certs` directory. I included this so that the old way of passing in certificates (i.e. modifying the files in `certs`) continues to be supported.

Let me know if that makes sense and/or if you'd like to see any changes.